incubator/declarative-linking/src/test/java/org/glassfish/jersey/linking/MethodProcessorTest.java

```diff
@@ -0,0 +1,905 @@
+        }
+
+        @Override
+        public Throwable getMappedThrowable() {
+            throw new UnsupportedOperationException("Not supported yet.");
+        }
+
+        @Override
+        public List<MatchResult> getMatchedResults() {
+            throw new UnsupportedOperationException("Not supported yet.");
+        }
+
+        @Override
+        public List<UriTemplate> getMatchedTemplates() {
+            throw new UnsupportedOperationException("Not supported yet.");
+        }
+
+        @Override
+        public List<PathSegment> getPathSegments(String name) {
+            throw new UnsupportedOperationException("Not supported yet.");
+        }
+
+        @Override
+        public List<PathSegment> getPathSegments(String name, boolean decode) {
+            throw new UnsupportedOperationException("Not supported yet.");
+        }
+
+        @Override
+        public List<RuntimeResource> getMatchedRuntimeResources() {
+            throw new UnsupportedOperationException("Not supported yet.");
+        }
+
+        @Override
+        public ResourceMethod getMatchedResourceMethod() {
+            throw new UnsupportedOperationException("Not supported yet.");
+        }
+
+        @Override
+        public Resource getMatchedModelResource() {
+            throw new UnsupportedOperationException("Not supported yet.");
+        }
+
+        @Override
+        public List<ResourceMethod> getMatchedResourceLocators() {
+            throw new UnsupportedOperationException("Not supported yet.");
+        }
+
+        @Override
+        public List<Resource> getLocatorSubResources() {
+            throw new UnsupportedOperationException("Not supported yet.");
+        }
+
+        @Override
+        public URI resolve(URI uri) {
+            throw new UnsupportedOperationException("Not supported yet.");
+        }
+
+        @Override
+        public URI relativize(URI uri) {
+            throw new UnsupportedOperationException("Not supported yet.");
+        }
+    };
+
+    private final ResourceMappingContext mockRmc = new ResourceMappingContext() {
+
+        @Override
+        public ResourceMappingContext.Mapping getMapping(Class<?> resource) {
+            return null;
+        }
+    };
+
+
+    private final static String TEMPLATE_A = "foo";
+
+    public static class TestClassD {
+        private String res1;
+        private URI res2;
+        
+        @InjectLink(value = TEMPLATE_A, style = InjectLink.Style.RELATIVE_PATH)
+        public String getRes1() {
+        	return res1;
+        }
+        
+        public void setRes1(String res1) {
+        	this.res1 = res1;
+        }
+
+        @InjectLink(value = TEMPLATE_A, style = InjectLink.Style.RELATIVE_PATH)
+        public URI getRes2() {
+        	return res2;
+        }
+        
+        public void setRes2(URI uri) {
+        	this.res2 = uri;
+        }
+    }
+
+    @Test
+    public void testProcessLinks() {
+        System.out.println("Links");
```

Comment:
I know, that we have system.outs in some other tests as well, but I would prefer not to add new ones. We try to get rid of those, when we are occasionally changing some tests, so would be nice to use a jdk logger:
`private static final Logger LOG = Logger.getLogger(FooBarTest.class.getName())` and then simply `LOG.info()`, `LOG.fine()`, etc. 
Would be nice to have those "look, I am entering a method" log either removed completely or logged *bellow* the info level (even though I admit that you can definitely find some contra-examples in our existing code.